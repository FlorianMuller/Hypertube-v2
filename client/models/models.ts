import { AxiosResponse, AxiosError, Canceler } from "axios";

export interface Locale {
  locale: string;
  setLocale: (locale: string) => void;
}

export type Method = "get" | "put" | "post" | "patch" | "delete";
export type Headers = { [key: string]: string } | undefined;
export type Data = object | undefined;

export interface UseApiOption {
  method?: Method;
  headers?: Headers;
  data?: Data;
  hotReload?: boolean;
  validateStatus?: (status: number) => boolean;
}

export interface UseApiReturn<T, E> {
  callApi: (freshData?: Data) => Promise<void>;
  loading: boolean;
  res: AxiosResponse<T>;
  resData: T;
  error: AxiosError<E>;
  setUrl: (url: string) => void;
  setMethod: (method: Method) => void;
  setHeaders: (headers: Headers) => void;
  setData: (data: Data) => void;
  cancelAllRequests: Canceler;
}

export interface ApiAuthResponse {
  validToken: boolean;
  error?: string;
}

export interface AuthInfo {
  username: string;
  password: string;
  [key: string]: string;
}

export interface Comment {
  movieName: string;
  date: number;
  name: string;
  body: string;
  stars: number;
  _id: number;
}

export interface Review {
  id: string;
  name: string;
  date: number;
  stars: number;
  body: string;
}

export interface Filters {
  query: string;
  collections: string[];
  year: number;
  minRating: number;
}

export interface Movie {
  id: string;
  title: string;
  cover: string;
  year: number;
  summary: string;
  genres: string[];
  rating: number;
  runtime: number;
}

export interface ApiSearchReponse {
  movies: Movie[];
  nextPage: boolean;
}

export type CustomSnackbarVariant =
  | "success"
  | "warning"
  | "error"
  | "info"
  | "default";
