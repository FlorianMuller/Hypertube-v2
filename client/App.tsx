import * as React from "react";

import { useIntl } from "react-intl";

interface Props {
  setLocale: (locale: string) => void;
}

const App = ({ setLocale }: Props) => {
  const t = useIntl();

  return <div>{t.formatMessage({ id: "helloworld" })}</div>;
};

export default App;
