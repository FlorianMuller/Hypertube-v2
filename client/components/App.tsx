import React, { ReactElement } from "react";
import { IntlProvider } from "react-intl";
import { hot } from "react-hot-loader/root";
import { Router, Switch, Route, Link } from "react-router-dom";

import CssBaseline from "@material-ui/core/CssBaseline";
import { MuiThemeProvider, createMuiTheme } from "@material-ui/core/styles";
import history from "../helpers/history";

import enTranslation from "../translations/en.json";
import frTranslation from "../translations/fr.json";

import CustomRoute from "./Routes/CustomRoute";
import Layout from "./Layout";
import Authentication from "./Authentication/Authentication";
import Home from "./Home/Home";
import Movie from "./Movie/Movie";
import Search from "./Search";

export const theme = createMuiTheme({
  palette: {
    type: "dark"
  }
});

const messages: Record<string, Record<string, string>> = {
  en: enTranslation,
  fr: frTranslation
};

const App = (): ReactElement => {
  const [locale, setLocale] = React.useState("en");

  return (
    <MuiThemeProvider theme={theme}>
      <CssBaseline />
      <IntlProvider locale={locale} messages={messages[locale]}>
        <Layout locale={locale} setLocale={setLocale}>
          <Router history={history}>
            <Switch>
              <CustomRoute
                exact
                path="/sign-in"
                component={Authentication}
                requireAuth={false}
              />
              <CustomRoute
                exact
                path="/sign-up"
                component={Authentication}
                requireAuth={false}
              />
              <CustomRoute
                exact
                path="/reset-password"
                component={Authentication}
                requireAuth={false}
              />
              <Route path="/search" component={Search} />
              <CustomRoute path="/movie" component={Movie} requireAuth />
              <Route exact path="/" component={Home} />
              <Route
                component={(): ReactElement => (
                  <div>
                    Not found, go to <Link to="/">root</Link>
                  </div>
                )}
              />
            </Switch>
          </Router>
        </Layout>
      </IntlProvider>
    </MuiThemeProvider>
  );
};

export default hot(App);
