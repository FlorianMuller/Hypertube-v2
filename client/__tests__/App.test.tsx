import * as React from "react";
import Adapter from "enzyme-adapter-react-16";
import EnzymeToJson from "enzyme-to-json";
import { mount, configure } from "enzyme";
import App from "../components/App";

import { UseApiReturn, ApiAuthResponse } from "../models/models";

configure({ adapter: new Adapter() });

// `loading: true` to avoid mounting all <Home> or <SignIn> component
jest.mock("../hooks/useApi", () => (): UseApiReturn<
  ApiAuthResponse,
  ApiAuthResponse
> => ({
  callApi: jest.fn(),
  loading: true,
  res: null,
  resData: null,
  error: null,
  setUrl: jest.fn(),
  setMethod: jest.fn(),
  setHeaders: jest.fn(),
  setData: jest.fn(),
  cancelAllRequests: jest.fn()
}));

jest.mock("../helpers/history", () => {
  // require after code makes the disable required
  // eslint-disable-next-line
  const { createBrowserHistory } = require("history");

  const history = createBrowserHistory({
    keyLength: 0
  });

  return history;
});

describe("App", () => {
  it("renders correctly", () => {
    const domNode = mount(<App />);
    expect(EnzymeToJson(domNode)).toMatchSnapshot();
  });
});
