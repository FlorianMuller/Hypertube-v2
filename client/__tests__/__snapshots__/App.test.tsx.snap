// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div>
  <div
    className="makeStyles-root-1"
  >
    <header
      className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary"
    >
      <div
        className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <button
          aria-label="open drawer"
          className="MuiButtonBase-root MuiIconButton-root makeStyles-menuButton-2 MuiIconButton-colorInherit MuiIconButton-edgeStart"
          disabled={false}
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
          </span>
        </button>
        <h6
          className="MuiTypography-root makeStyles-title-3 MuiTypography-h6 MuiTypography-noWrap"
        >
          HyperTube
        </h6>
        <div
          className="makeStyles-search-4"
        >
          <div
            className="makeStyles-searchIcon-5"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
          <div
            className="MuiInputBase-root makeStyles-inputRoot-6"
            onClick={[Function]}
          >
            <input
              aria-label="search"
              className="MuiInputBase-input makeStyles-inputInput-7"
              onAnimationStart={[Function]}
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              placeholder="Search..."
              type="text"
            />
          </div>
        </div>
      </div>
    </header>
  </div>
  <div>
    <a
      href="/"
      onClick={[Function]}
    >
      Not authenticated
    </a>
    <div
      className="MuiPaper-root MuiPaper-elevation1 makeStyles-page-120 MuiPaper-rounded"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
      >
        <div
          className="MuiGrid-root makeStyles-titles-121 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
        >
          <div
            className="MuiGrid-root MuiGrid-item"
          >
            <h3
              className="MuiTypography-root MuiTypography-h3"
            >
              Who are you ?
              <span
                aria-label="Eyes"
                role="img"
              >
                 ðŸ‘€
              </span>
            </h3>
          </div>
          <div
            className="MuiGrid-root makeStyles-subtitle-122 MuiGrid-item"
          >
            <h6
              className="MuiTypography-root MuiTypography-subtitle1"
            >
              To access all our film, you're gonna have to register
              <span
                aria-label="Eyes"
                role="img"
              >
                 ðŸ¤­
              </span>
            </h6>
          </div>
        </div>
        <form
          onSubmit={[Function]}
        >
          <div
            className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
          >
            <div
              className="MuiGrid-root makeStyles-item-123 MuiGrid-item"
            >
              <div
                className="MuiFormControl-root MuiTextField-root makeStyles-textInput-124"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
                  data-shrink={false}
                >
                  Your little username
                </label>
                <div
                  className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={true}
                    className="MuiInputBase-input MuiFilledInput-input"
                    disabled={false}
                    maxLength={30}
                    name="username"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="text"
                    value=""
                  />
                </div>
                
              </div>
            </div>
            <div
              className="MuiGrid-root makeStyles-item-123 MuiGrid-item"
            >
              <div
                className="MuiFormControl-root MuiTextField-root makeStyles-textInput-124"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
                  data-shrink={false}
                >
                  Your top secret password
                </label>
                <div
                  className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInputBase-input MuiFilledInput-input"
                    disabled={false}
                    maxLength={1028}
                    name="password"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="password"
                    value=""
                  />
                </div>
                
              </div>
            </div>
            <div
              className="MuiGrid-root makeStyles-item-123 MuiGrid-item"
            >
              <div
                className="MuiFormControl-root MuiTextField-root makeStyles-textInput-124"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
                  data-shrink={false}
                >
                  Your email (so we can spam you)
                </label>
                <div
                  className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInputBase-input MuiFilledInput-input"
                    disabled={false}
                    maxLength={100}
                    name="email"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="text"
                    value=""
                  />
                </div>
                
              </div>
            </div>
            <div
              className="MuiGrid-root makeStyles-item-123 MuiGrid-item"
            >
              <div
                className="MuiFormControl-root MuiTextField-root makeStyles-textInput-124"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
                  data-shrink={false}
                >
                  Your first name
                </label>
                <div
                  className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInputBase-input MuiFilledInput-input"
                    disabled={false}
                    maxLength={30}
                    name="firstName"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="text"
                    value=""
                  />
                </div>
                
              </div>
            </div>
            <div
              className="MuiGrid-root makeStyles-item-123 MuiGrid-item"
            >
              <div
                className="MuiFormControl-root MuiTextField-root makeStyles-textInput-124"
              >
                <label
                  className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-filled"
                  data-shrink={false}
                >
                  Your last name
                </label>
                <div
                  className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-formControl"
                  onClick={[Function]}
                >
                  <input
                    aria-invalid={false}
                    autoFocus={false}
                    className="MuiInputBase-input MuiFilledInput-input"
                    disabled={false}
                    maxLength={30}
                    name="lastName"
                    onAnimationStart={[Function]}
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    required={false}
                    type="text"
                    value=""
                  />
                </div>
                
              </div>
            </div>
            <div
              className="MuiGrid-root makeStyles-item-123 MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center"
            >
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <p
                  className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
                >
                  Your handsome face
                  <span
                    aria-label="Arm taking a selfie"
                    role="img"
                  >
                     ðŸ¤³
                  </span>
                </p>
              </div>
              <div
                className="MuiGrid-root MuiGrid-item"
              >
                <label
                  htmlFor="raised-button-file"
                >
                  <input
                    accept="image/*"
                    id="raised-button-file"
                    multiple={true}
                    name="picture"
                    onChange={[Function]}
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                    type="file"
                  />
                  <span
                    aria-disabled={false}
                    className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary"
                    onBlur={[Function]}
                    onDragLeave={[Function]}
                    onFocus={[Function]}
                    onKeyDown={[Function]}
                    onKeyUp={[Function]}
                    onMouseDown={[Function]}
                    onMouseLeave={[Function]}
                    onMouseUp={[Function]}
                    onTouchEnd={[Function]}
                    onTouchMove={[Function]}
                    onTouchStart={[Function]}
                    role="button"
                    tabIndex={0}
                  >
                    <span
                      className="MuiButton-label"
                    >
                      <span
                        className="MuiButton-startIcon MuiButton-iconSizeMedium"
                      >
                        <svg
                          aria-hidden="true"
                          className="MuiSvgIcon-root"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96zM14 13v4h-4v-4H7l5-5 5 5h-3z"
                          />
                        </svg>
                      </span>
                      Upload
                    </span>
                  </span>
                </label>
              </div>
            </div>
            <div
              className="MuiGrid-root makeStyles-item-123 MuiGrid-item"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
                disabled={false}
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="submit"
              >
                <span
                  className="MuiButton-label"
                >
                  I register on this awesome website
                </span>
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
    <div>
      Sign In
    </div>
    <div>
      Resset password
    </div>
  </div>
</div>
`;
